ios/FluentUI/SegmentedControl/SegmentedControl.swift

```diff
@@ -474,29 +508,37 @@ open class SegmentedControl: UIControl {
         return button
     }
 
-    private func createSwitchButton(withTitle title: String) -> PillButton {
-        let pillStyle = style == .onBrandPill ? PillButtonStyle.onBrand : PillButtonStyle.primary
-        let button = PillButton(pillBarItem: PillButtonBarItem(title: title), style: pillStyle)
+    private func createSwitchButton(withTitle title: String) -> UIButton {
+        let button = UIButton()
         button.setTitle(title, for: .normal)
         button.accessibilityLabel = title
-        button.largeContentTitle = button.titleLabel?.text
+        button.largeContentTitle = title
         button.showsLargeContentViewer = true
+        button.titleLabel?.font = style.segmentTextFont
         button.addTarget(self, action: #selector(handleButtonTap(_:)), for: .touchUpInside)
+        button.contentEdgeInsets = Constants.pillButtonInsets
 
-        // set the pill background to transparent to let the selection view show through
-        let clear = UIColor.clear
-        button.customBackgroundColor = clear
-        button.customSelectedBackgroundColor = clear
+        return button
+    }
 
-        if let customButtonTextColor = self.customSegmentedControlButtonTextColor {
-            button.customTextColor = customButtonTextColor
-        }
+    private func addMaskedPillLabel(over button: UIButton, at index: Int) {
+        let maskedLabel = UILabel()
+        maskedLabel.text = button.currentTitle
+        maskedLabel.font = style.segmentTextFont
+        maskedLabel.translatesAutoresizingMaskIntoConstraints = false
+        pillMaskedLabelsContainerView.addSubview(maskedLabel)
+        pillMaskedLabels.insert(maskedLabel, at: index)
 
-        if let customSelectedButtonTextColor = self.customSelectedSegmentedControlButtonTextColor {
-            button.customSelectedTextColor = customSelectedButtonTextColor
+        var constraints = [NSLayoutConstraint]()
```

Comment:
you shouldn't need this variable at all right? 
`            NSLayoutConstraint.activate([
                buttonTitle.leadingAnchor.constraint(equalTo: maskedLabel.leadingAnchor),
                buttonTitle.trailingAnchor.constraint(equalTo: maskedLabel.trailingAnchor),
                buttonTitle.topAnchor.constraint(equalTo: maskedLabel.topAnchor),
                buttonTitle.bottomAnchor.constraint(equalTo: maskedLabel.bottomAnchor)
                ])`